modules/social_features/social_user_export/src/ExportUser.php

```diff
@@ -89,20 +100,42 @@ public static function exportUserOperation(UserInterface $entity, array &$contex
       $last_access = t('never');
     }
 
+    /** @var \Drupal\profile\ProfileStorageInterface $storage */
+    // Check if entity type 'profile' exists.
+    $storage = \Drupal::entityTypeManager()->getStorage('profile');
+    if (!empty($storage)) {
+      $user_profile = $storage->loadByUser($entity, 'profile', TRUE);
+    }
+    else {
+      $user_profile = NULL;
+    }
+
     // Add row.
     $row = [
       $entity->id(),
       $entity->uuid(),
-      social_user_export_first_name($entity),
-      social_user_export_last_name($entity),
+      _social_user_export_profile_get_field_value('field_profile_first_name', $user_profile),
+      _social_user_export_profile_get_field_value('field_profile_last_name', $user_profile),
       $entity->getAccountName(),
       $entity->getDisplayName(),
       $entity->getEmail(),
       $last_login,
       $last_access,
       \Drupal::service('date.formatter')->format($entity->getCreatedTime(), 'custom', 'Y/m/d - H:i'),
       !empty($status[0]['value']) ? t('Active') : t('Blocked'),
-      implode(', ', $roles),
+      _social_user_export_profile_get_address('field_profile_address', 'country_code', $user_profile),
+      _social_user_export_profile_get_address('field_profile_address', 'administrative_area', $user_profile),
+      _social_user_export_profile_get_address('field_profile_address', 'locality', $user_profile),
+      _social_user_export_profile_get_address('field_profile_address', 'postal_code', $user_profile),
+      _social_user_export_profile_get_address('field_profile_address', 'address_line1', $user_profile),
+      _social_user_export_profile_get_address('field_profile_address', 'address_line2', $user_profile),
+      _social_user_export_profile_get_field_value('field_profile_phone_number', $user_profile),
+      _social_user_export_profile_get_field_value('field_profile_organization', $user_profile),
+      _social_user_export_profile_get_field_value('field_profile_function', $user_profile),
+      _social_user_export_profile_get_taxonomy_field_value('field_profile_expertise', $user_profile),
+      _social_user_export_profile_get_taxonomy_field_value('field_profile_interests', $user_profile),
+      _social_user_export_profile_get_taxonomy_field_value('field_profile_profile_tag', $user_profile),
```

Comment:
@jaapjan profile tag module is optional right? Maybe put a `moduleExists` around it just like we did for private message.

Reply:
I also thought that, but social_tagging (for content) is optional. This field is always there, since it is delivered with `social_user` feature. If there are no profile tag taxonomy terms then the feature will just not do anything.